internal/terraform/terraform_test.go

```diff
@@ -170,24 +170,24 @@ func testSetResourceInstanceTainted(module *states.Module, resource, attrsJson,
 }
 
 func testProviderFuncFixed(rp providers.Interface) providers.Factory {
-	return func() (providers.Interface, error) {
-		if p, ok := rp.(*MockProvider); ok {
-			// make sure none of the methods were "called" on this new instance
-			p.GetProviderSchemaCalled = false
-			p.ValidateProviderConfigCalled = false
-			p.ValidateResourceConfigCalled = false
-			p.ValidateDataResourceConfigCalled = false
-			p.UpgradeResourceStateCalled = false
-			p.ConfigureProviderCalled = false
-			p.StopCalled = false
-			p.ReadResourceCalled = false
-			p.PlanResourceChangeCalled = false
-			p.ApplyResourceChangeCalled = false
-			p.ImportResourceStateCalled = false
-			p.ReadDataSourceCalled = false
-			p.CloseCalled = false
-		}
+	if p, ok := rp.(*MockProvider); ok {
```

Comment:
This has always been a very delicate an unreliable aspect of the mock provider. Resetting these on each factory call assumed that the same instance was used for an entire operation which isn't always true, and the new external caching makes that all the more apparent. We settle on making the provider reusable, but requiring a new closure for each context if you want to use these fields.